### PR TITLE
feat(rust-consumer): Move code from rust arroyo to a separate crate

### PR DESCRIPTION
**Pull request by lynnagara**
We don't want the arroyo code to be in the root of the Snuba project, move it to a separate crate

### CODE REVIEW

**Review by untitaker — COMMENTED** [review]
for more proper isolation, do a separate crate too

**Comment by lynnagara** [comment]
@untitaker Update rust_arroyo to a crate. Lmk if this setup looks right.

File: rust_snuba/rust_arroyo/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[package]
```
Comment (fpacifici) [inline rust_snuba/rust_arroyo/Cargo.toml:1]: Do you plan to actually publish a crate from rust_arroyo and import that eventually or do you plan to vendor rust_arroyo into snuba permanently ?

The first is better.

Comment (lynnagara) [reply rust_snuba/rust_arroyo/Cargo.toml:1]: I think the first is better in the long term and the second is better in the short term. I would hold off publishing a crate until we have a much better idea of what that library would look like. Right now we are unsure and prioritising for faster iteration.

**Review by dbanda — APPROVED** [review]

**Pull request merged**